docs/decision_monorepo_tooling.md

```diff
@@ -0,0 +1,30 @@
+# Decision Document: Monorepo Adoption
+
+## Overview
+
+The decision to adopt a monorepo structure for our projects is driven by the need for improved dependency management, task performance, and a streamlined release process. This document outlines the motivation and decision points.
+
+## Motivation for Monorepo
+
+The decision to adopt a monorepo is driven by the following considerations:
+
+- **Simplified Dependency Management:** Using internal dependencies within the monorepo simplifies version handling and enhances the development experience.
+- **Improved Task Management and Performance:** Enhancing task management and improving performance through effective caching mechanisms during development.
+- **Streamlined Release Process:** Preparing packages for release with a standardized and efficient workflow.
+
+---
+
+## Decision Points
+
+To achieve these goals, the following decisions have been made:
+
+1. **Dependency Management:**
+
+   - Internal dependencies within the monorepo will be marked with `*` to streamline development processes.
+
+2. **Tool Adoption:**
+
+   - Adoption of `turborepo` is proposed to optimize task management and leverage caching for performance improvements.
+
+3. **Release Management:**
+   - The `changeset` CLI tool will be used to manage versioning and changelogs for packages within the monorepo. It will streamline the process of tracking changes, updating versions, and generating changelogs for releases.
```

Comment:
I think it would be nice to make it consistent with other points so same new lines in between like in point 1 and 2